openapi/openapi.yml

```diff
@@ -268,120 +268,142 @@ components:
               items:
                 $ref: "#/components/schemas/SpocFeedItem"
 
-    FeedItem:
+    Shim:
       type: object
+      additionalProperties: false
       properties:
-        id:
-          type: integer
-          example: 30295
-        title:
+        click:
           type: string
-          example: Why driving is hard—even for AIs
-        domain:
+          example: "1234123asdf4tYadsfQ,xY-01BU12"
+        impression:
           type: string
-          example: arstechnica.com
-        image:
+          example: "a0c3943asdf4tYadsf300,xY-01BU9aadc"
+        delete:
           type: string
-          example: "https://img-getpocket.cdn.mozilla.net/direct?url=https%3A%2F%2Fcdn.arstechnica.net%2Fwp-content%2Fuploads%2F2018%2F12%2FGettyImages-898172236-640x427.jpg&resize=w450"
-        raw_image_src:
+          example: "fdea123asdf4tYadsf1000,xY-01BUa654"
+        save:
           type: string
-          example: "https://cdn.arstechnica.net/wp-content/uploads/2018/12/GettyImages-898172236-640x427.jpg"
-        published_timestamp:
+          example: "4567123asdf4tYadsfQcda,xY-01BU123"
+
+    Caps:
+      type: object
+      additionalProperties: false
+      required:
+        - lifetime
+        - flight
+        - campaign
+      properties:
+        lifetime:
           type: integer
-          description: Unix timestamp of when this item was published
-          example: 1543471200
-        item_score:
-          type: number
-          example: 0.98
-        parameter_set:
+          example: 50
+        flight:
+          type: object
+          additionalProperties: false
+          required:
+            - count
+            - period
+          properties:
+            count:
+              type: integer
+              example: 10
+            period:
+              type: integer
+              description: Period in seconds
+              example: 86400
+        campaign:
+          type: object
+          additionalProperties: false
+          required:
+            - count
+            - period
+          properties:
+            count:
+              type: integer
+              example: 10
+            period:
+              type: integer
+              description: Period in seconds
+              example: 86400
+
+    SpocFeedItem:
+      type: object
+      additionalProperties: false
```

Comment:
i could have made a list of required fields here, but wasn't sure which ones could be empty values in kevel so didn't want to add any incorrect assumptions. for now this validates what fields are allowed and excludes any fields not listed